discovery/azure/azure.go

```diff
@@ -364,35 +365,29 @@ func (d *Discovery) refresh() (tg *targetgroup.Group, err error) {
 					return
 				}
 
-				if networkInterface.Properties == nil {
```

Comment:
@simonpasquier @sylr Now that I look at this again. This is definitely needed. I will add this if check back.